pkg/cmd/run.go

```diff
@@ -157,10 +162,6 @@ doppler run --mount secrets.json -- cat secrets.json`,
 		}
 
 		if cmd.Flags().Changed("only-secrets") {
-			if len(secretsToInclude) == 0 {
-				utils.HandleError(fmt.Errorf("you must specify secrets when using --only-secrets"))
-			}
-
 			var err error
 			noExitOnMissingIncludedSecrets := cmd.Flags().Changed("no-exit-on-missing-only-secrets")
 			dopplerSecrets, err = controllers.SelectSecrets(dopplerSecrets, secretsToInclude)
```

Comment:
With the new functionality of downloading only specified secrets, we can re-organize this block and rework the `SelectSecrets` function. We no longer need to filter on the client side since the server only returns the specified secrets.

We might want to refactor `SelectSecrets`, remove the filter logic, rename the function to something like `detectedMissingSecrets` which takes the `secretsToInclude` and the result of the fetch call https://github.com/DopplerHQ/cli/blob/8ac1e7cfa3183c3885ed9bd2a99976883320016a/pkg/controllers/secrets.go#L243-L259

Maybe something along the lines of:
```go
func detectedMissingIncludedSecrets(secrets map[string]string, secretsToInclude []string) (map[string]string, error) {
	nonExistentSecretNames := []string{}

	for _, secretName := range secretsToInclude {
		if _, found := secrets[secretName]; !found {
			nonExistentSecretNames = append(nonExistentSecretNames, secretName)
		}
	}

	var err error
	if len(nonExistentSecretNames) > 0 {
		err = fmt.Errorf("the following secrets you are trying to include do not exist in your config:\n- %v", strings.Join(nonExistentSecretNames, "\n- "))
	}

	return secrets, err
}
```


```go
var dopplerSecrets = fetchSecrets(localConfig, enableCache, fallbackOpts, metadataPath, 
nameTransformer, dynamicSecretsTTL, format, secretsToInclude)
// ...
if cmd.Flags().Changed("only-secrets") {
	res, err := detectMissingIncludedSecrets(dopplerSecrets, secretsToInclude)
        /...
}
```

Reply:
Good call, we can definitely remove this client-side filtering logic. Will refactor.

Reply:
Updates this logic to only search for missing secrets